src/GitVersionCore/BranchConfigurationCalculator.cs

```diff
@@ -27,11 +27,20 @@ public static BranchConfig GetBranchConfiguration(GitVersionContext context, Bra
                 ConfigurationProvider.ApplyBranchDefaults(context.FullConfiguration, matchingBranches, "", new List<string>());
             }
 
-            return matchingBranches.Increment == IncrementStrategy.Inherit ?
-                InheritBranchConfiguration(context, targetBranch, matchingBranches, excludedInheritBranches) :
-                matchingBranches;
+            if (matchingBranches.Increment == IncrementStrategy.Inherit)
```

Comment:
This shouldn't happen now because we filter the conditions which would cause this code to be executed. But still, I think it's worth keeping in just in case